app/gui/src/controller/ide.rs

```diff
@@ -150,18 +157,44 @@ pub trait ManagingProjectAPI {
     /// and then for the project opening.
     fn open_project_by_name(&self, name: String) -> BoxFuture<FallibleResult> {
         async move {
-            let projects = self.list_projects().await?;
-            let mut projects = projects.into_iter();
-            let project = projects.find(|project| project.name.as_ref() == name);
-            let uuid = project.map(|project| project.id);
-            if let Some(uuid) = uuid {
-                self.open_project(uuid).await
-            } else {
-                Err(ProjectNotFound(name).into())
+            let project_id = self.find_project(&ProjectToOpen::Name(name.into())).await?;
+            self.open_project(project_id).await
+        }
+        .boxed_local()
+    }
+
+    /// Open a project by name or ID. If no project with the given name exists, it will be created.
+    fn open_or_create_project(&self, project_to_open: ProjectToOpen) -> BoxFuture<FallibleResult> {
+        async move {
+            match self.find_project(&project_to_open).await {
+                Ok(project_id) => self.open_project(project_id).await,
+                Err(error) =>
+                    if let ProjectToOpen::Name(name) = project_to_open {
+                        info!("Attempting to create project with name {}", name);
```

Comment:
1. Logs should be nicely formatted. See other logs in this file, they quote names.
2. Logs should use real English sentences, this one is missing dot. 

Please check all other places in this PR for similar issues.

Reply:
This is just a line from the old code I added some words to. In the codebase I see a mix of unquoted, single quotes, backticks and `[]` (in the engine) being used. I couldn't find other logs in this file like you mentioned, so do you have a preference?

Reply:
I think the nicest way of doing it would be to write `Attempting to create project with name '{name}'.` :)